src/openloco/windows/music_selection.cpp

```diff
@@ -0,0 +1,236 @@
+            string_ids::music_get_me_to_gladstone_bay,
+            string_ids::music_sandy_track_blues,
+        };
+
+        auto config = config::get();
+
+        uint16_t y = 0;
+        for (uint16_t i = 0; i < window->row_count; i++)
+        {
+            string_id text_colour_id = string_ids::white_stringid2;
+
+            // Draw hovered track
+            if (i == window->row_hover)
+            {
+                gfx::draw_rect(dpi, 0, y, 800, rowHeight - 1, 0x2000030);
+                text_colour_id = string_ids::wcolour2_stringid2;
+            }
+
+            // Draw checkbox.
+            gfx::fill_rect_inset(dpi, 2, y, 11, y + 10, window->colours[1], 0xE0);
+
+            // Draw checkmark if track is enabled.
+            if (config.enabled_music[i])
+                gfx::draw_string_494B3F(*dpi, 2, y, window->colours[1], string_ids::wcolour2_stringid2, (void*)&string_ids::checkmark);
+
+            // Draw track name.
+            gfx::draw_string_494B3F(*dpi, 15, y, window->colours[1], text_colour_id, (void*)&music_ids_to_string_id[i]);
+
+            y += rowHeight;
+        }
+    }
+
+    // 0x004C176C
+    static void get_scroll_size(ui::window* window, uint32_t scrollIndex, uint16_t* scrollWidth, uint16_t* scrollHeight)
+    {
+        *scrollHeight = rowHeight * numTracks;
+    }
+
+    // 0x004C1757
+    static void on_mouse_up(ui::window* window, widget_index widgetIndex)
+    {
+        switch (widgetIndex)
+        {
+            case widx::close:
+                WindowManager::close(window->type);
+                break;
+        }
+    }
+
+    // 0x004C1799
+    static void on_scroll_mouse_down(ui::window* window, int16_t x, int16_t y, uint8_t scroll_index)
+    {
+        uint16_t currentTrack = y / rowHeight;
+        if (currentTrack > window->row_count)
+            return;
+
+        auto& config = config::get();
+
+        // Toggle the track in question.
+        config.enabled_music[currentTrack] = !config.enabled_music[currentTrack];
+
+        // Are any tracks enabled?
+        bool anyEnabled = false;
+        for (uint8_t i = 0; i < numTracks; i++)
+            anyEnabled |= config.enabled_music[i];
+
+        // Ensure at least this track is enabled.
+        if (!anyEnabled)
+            config.enabled_music[currentTrack] = true;
+
+        config::write();
+        call(0x048AA0C);
+        window->invalidate();
+    }
+
+    // 0x004C1771
+    static void on_scroll_mouse_over(ui::window* window, int16_t x, int16_t y, uint8_t scroll_index)
+    {
+        uint16_t currentTrack = y / rowHeight;
+        if (currentTrack > window->row_count || currentTrack == window->row_hover)
+            return;
+
+        window->row_hover = currentTrack;
+        window->invalidate();
+    }
+
+    // 0x004C17E3
+    static void on_update(window* window)
+    {
+        auto optionsWindow = WindowManager::find(WindowType::options);
+        if (optionsWindow == nullptr || optionsWindow->current_tab != 2)
```

Comment:
can we add a constant for this 2 in a header in the options namespace?

Reply:
Sure… Yet another one for the WindowManager header, or should we introduce a WidgetReference.h for such matters (or indeed a better name, if you can think of one)?

Reply:
I think it'd be fine (for now at least) in the options namespace, with a compile-time assert in options.cpp